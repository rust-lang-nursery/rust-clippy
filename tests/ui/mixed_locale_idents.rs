#![warn(clippy::mixed_locale_idents)]
#[allow(dead_code)]

// Russian `о`.
pub struct Blоck;

fn main() {
    let black_чёрный_黒い_काला = "good luck hand-writing it";

    // Should not spawn the lint, as it represents valid sequence in a single locale.
    let nutzer_zähler = "user counter";
}
