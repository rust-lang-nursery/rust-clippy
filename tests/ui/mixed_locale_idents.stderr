error: identifier part `Blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:7:16
   |
LL |     pub struct Blоck;
   |                ^^^^^
   |
   = note: `-D clippy::mixed-locale-idents` implied by `-D warnings`

error: identifier part `BLОCK` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:8:15
   |
LL |     pub const BLОCK: u8 = 42;
   |               ^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:9:12
   |
LL |     pub fn blоck() {
   |            ^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:10:13
   |
LL |         let blоck = 42u8;
   |             ^^^^^

error: identifier part `Blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:14:16
   |
LL |     pub struct SomeBlоckIdentifier;
   |                ^^^^^^^^^^^^^^^^^^^

error: identifier part `BLОCK` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:15:15
   |
LL |     pub const SOME_BLОCK_IDENT: u8 = 42;
   |               ^^^^^^^^^^^^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:16:12
   |
LL |     pub fn some_blоck_fn() {
   |            ^^^^^^^^^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:17:13
   |
LL |         let some_blоck_var = 42u8;
   |             ^^^^^^^^^^^^^^

error: identifier part `Blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:19:16
   |
LL |     pub struct Some_BlоckIdent; // Mixed case
   |                ^^^^^^^^^^^^^^^

error: identifier part `Blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:23:16
   |
LL |     pub struct BlоckClоck;
   |                ^^^^^^^^^^

error: identifier part `BLОCK` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:24:15
   |
LL |     pub const BLОCK_CLОCK: u8 = 42;
   |               ^^^^^^^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:25:12
   |
LL |     pub fn blоck_clоck() {
   |            ^^^^^^^^^^^

error: identifier part `blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:26:13
   |
LL |         let blоck_clоck = 42u8;
   |             ^^^^^^^^^^^

error: identifier part `Blоck` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:31:16
   |
LL |     pub struct SomeБлокBlоck;
   |                ^^^^^^^^^^^^^

error: identifier part `Blоck看` contains confusables-only symbols from the following locales: Cyrillic
  --> $DIR/mixed_locale_idents.rs:36:16
   |
LL |     pub struct Blоck看;
   |                ^^^^^^^

error: aborting due to 15 previous errors

